website/versioned_docs/version-v1.x/workflow.md

```diff
@@ -0,0 +1,142 @@
+controls a unique private key - MACI does not specifically solve for this, but allows for customization on how this is configured.
+
+#### Vote
+
+Once a user has signed up with MACI, they are eligible to vote on open polls.
+
+To cast a vote, a user will bundle a few variables — including a public key, their vote option, their vote amount, and a few others — into what is called a "command". Then, the user signs the command with the public key they originally used to sign up. After that, the user encrypts the signature and command together so that it is now considered a "message". If the command is properly signed by the user’s MACI public key, then the message is considered valid will be counted in the final tally. Therefore, the MACI public key can be thought of as the user’s voting username, and the signature is the voting password. If they provide the correct signature, they can submit a vote.
+
+Before sending their vote on-chain, users encrypt their vote using a shared key that only the user and coordinator can know. This key scheme is designed so that every individual user shares a distinct key with the coordinator. This prevents any bribers from simply reading the transaction data to see which option a user voted for. It also allows the coordinator to decrypt user votes (so that they can tally the results).
+
+### 2) Closed
+
+A Poll is "closed", when the voting period ends. At this point, no users may submit any more votes. The Poll remains closed until the coordinator tallies the final results.
+
+At this point, the coordinator must process all the messages, tally the results, and publish the proofs on-chain.
+
+#### Process Messages
+
+Once the voting period has completed for a specific poll, the coordinator will use the MessageProcessor contract to first prove that they have correctly decrypted each message and applied them to correctly create an updated state tree. This state tree keeps an account of all the valid votes that should be counted. So, when processing the messages, the coordinator will not keep messages that are later overridden by a newer message inside the state tree. For example, if a user votes for option A, but then later sends a new message to vote for option B, the coordinator will only count the vote for option B.
+
+The coordinator must process messages in batches so that proving on chain does not exceed the data limit. The coordinator then creates a zk-SNARK proving their state tree correctly contains only the valid messages. Once the proof is ready, the coordinator calls `MessageProcessor.processMessages()`, providing a hash of the state tree and the zk-SNARK proof as an input parameters.
```

Comment:
should we add a link to the soliditydocs about `MessageProcessor.processMessages()`?